cmd/genji/shell/command.go

```diff
@@ -136,3 +141,193 @@ func displaySuggestions(in string) error {
 	fmt.Println()
 	return nil
 }
+
+func getDatabaseTransaction(db *genji.DB) (*database.Transaction, error) {
+	var dtx *database.Transaction
+	var err error
+	err = db.View(func(tx *genji.Tx) error {
+		d := tx.DB()
+		opts := &database.TxOptions{}
+		dtx, err = d.BeginTx(opts)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Println("BEGIN TRANSACTION")
+	fmt.Println()
+
+	return dtx, nil
+}
+
+func dumpTable(db *genji.DB, dtx *database.Transaction, table string) error {
+	// Get the given table.
+	t, err := dtx.GetTable(table)
+	if err != nil {
+		return err
+	}
+
+	// Get table info.
+	ti, err := t.Info()
+	if err != nil {
+		return err
+	}
+
+	fields := ti.FieldConstraints
+	if len(fields) == 0 {
+		fmt.Printf("CREATE TABLE %s;\n", table)
+	} else {
+		fmt.Printf("CREATE TABLE %s(", table)
+		for i, f := range fields {
+			fmt.Printf("%s %s", fields[i].Path, strings.ToUpper(fields[i].Type.String()))
+			if f.IsPrimaryKey {
+				fmt.Printf(" PRIMARY KEY")
+			}
+
+			if f.IsNotNull {
+				fmt.Printf(" IS NOT NULL")
```

Comment:
This is not correct, this should be `NOT NULL`.
Also, please test the dump manually to see if everything is correct

Reply:
Well seen.